### ISSUE TITLE
CLI deploy/deployment destroy deletes job in Nomad

### ISSUE THREAD

**Issue by siccous** [hashicorp/waypoint#3316]
When I invoke _waypoint deploy_ (it destroys old deployments during the release phase) or _waypoint deployment destroy_ then it also deletes the job in Nomad:

![image](https://user-images.githubusercontent.com/7812885/166918080-61f17246-1097-47d8-9577-5a8eca9cb0a5.png)

**Steps to Reproduce**
1. waypoint deployment list
![image](https://user-images.githubusercontent.com/7812885/166918619-1195cc82-0876-4f24-ae72-c9fcf0da75cf.png)

2. job tim is properly running in Nomad
3. waypoint deployment destroy 267
4. job tim is deleted
![image](https://user-images.githubusercontent.com/7812885/166918910-f149a3c5-d277-4dcb-b2e3-7955cbd35aec.png)

**Waypoint Platform Versions**

* Waypoint CLI Version: 0.8.1
* Waypoint Server Platform and Version: Nomad 0.8.1
* Waypoint Plugin: nomad-jobspec

**Comment by paladin-devops** [comment]
Hey @siccous! Thanks for opening this issue. What's happening here is that the `nomad-jobspec-canary` releaser is actually being invoked as the default releaser plugin for the `nomad-jobspec` deployer, even though you may not have the canary releaser configured in your `waypoint.hcl`. This is significant because Waypoint's default behavior for releases is to "prune" old deployments. For `nomad-jobspec`, a "mutable deployment" plugin, pruning results in deletion of the job, because a prune by Waypoint looks up the Nomad job by ID to delete it. The job ID does not change between old and new iterations though, which results in the deletion of the latest deployment.

This is [called out in the `nomad-jobspec-canary`](https://www.waypointproject.io/plugins/nomad#nomad-jobspec-canary-releasemanager) docs, but if you don't have that configured as a releaser, it stands to reason that you haven't needed to read through that documentation, and would not know this.

To address this, we're planning to refactor the canary plugin so that it's not invoked by default for the `nomad-jobspec` plugin.

**Comment by siccous** [comment]
@paladin-devops thanks for the clarification. I am aware that your PR with the canary releaser has been merged, but didn't notice it will be default :)

I've tried to read the docs about it again and I admit that I didn't fully understand it's function. For example my current jobspec contains this update stanza:

  update {
    max_parallel = 1
    canary = 1
    auto_revert = true
    auto_promote = true
  }

And it breaks as I described with the canary releaser, so what would be the proper way to configure this?
